receiver/kubeletstatsreceiver/internal/kubelet/metadata.go

```diff
@@ -133,6 +133,12 @@ func (m *Metadata) getContainerID(podUID string, containerName string) (string,
 					return stripContainerID(containerStatus.ContainerID), nil
 				}
 			}
+			for _, initContainerStatus := range pod.Status.InitContainerStatuses {
```

Comment:
Hm... Does this receiver even collect init container metric? I may be wrong, but don't think so... If not, maybe we there is a way to remove init containers from call chain and make sure they are not passed to `getContainerID` instead?

Reply:
> Does this receiver even collect init container metric? 

It does collect init container. The `/stats/summary` endpoint output does not differentiate initContainers from regular containers, so everything in the "container stats" will be processed, including init containers.

example MTS from the following init container:

```
[vagrant@control-plane ~]$ kubectl describe pod init-app-container
Name:         init-app-container
Namespace:    default
Priority:     0
Node:         localhost.localdomain/10.0.2.15
Start Time:   Tue, 02 Aug 2022 13:59:34 +0000
Labels:       app=initapp
Annotations:  <none>
Status:       Running
IP:           172.17.0.6
IPs:
  IP:  172.17.0.6
Init Containers:
  init-example:
    Container ID:  docker://0de200cb50c3b34df3f928cff7ed464a5a71a847f57df57262d6cd2ff00c6c08
    Image:         busybox:1.28
    Image ID:      docker-pullable://busybox@sha256:141c253bc4c3fd0a201d32dc1f493bcf3fff003b6df416dea4f41046e0f37d47
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      sleep 20
    State:          Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Tue, 02 Aug 2022 13:59:35 +0000
      Finished:     Tue, 02 Aug 2022 13:59:55 +0000
    Ready:          True
    Restart Count:  0
    Environment:    <none>
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-l2jtd (ro)
Containers:
  myapp-container:
    Container ID:  docker://9ca996396059fe22cac1146292c94b47f818ac068e7cc8e01e3d246fa0ea7c1c
    Image:         busybox:1.28
    Image ID:      docker-pullable://busybox@sha256:141c253bc4c3fd0a201d32dc1f493bcf3fff003b6df416dea4f41046e0f37d47
    Port:          <none>
    Host Port:     <none>
    Command:
      sh
      -c
      sleep 360000
    State:          Running
      Started:      Tue, 02 Aug 2022 13:59:55 +0000
```


```
us0-sfc -O CfhbPnRAcAM mb s -t mts 'container.id:*0de200cb50c3*' -l1
FZKTQx1AgAU
    app: "initapp"
    container.id: "0de200cb50c3b34df3f928cff7ed464a5a71a847f57df57262d6cd2ff00c6c08"
    host.name: "control-plane.minikube.internal"
    host_cpu_cores: "2"
    host_cpu_model: "Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz"
    host_kernel_name: "linux"
    host_kernel_release: "3.10.0-1160.36.2.el7.x86_64"
    host_kernel_version: "#1 SMP Wed Jul 21 11:57:15 UTC 2021"
    host_logical_cpus: "2"
    host_machine: "x86_64"
    host_mem_total: "3880236"
    host_physical_cpus: "2"
    host_processor: "x86_64"
    k8s.cluster.name: "minikubeSWAT"
    k8s.container.name: "init-example"
    k8s.namespace.name: "default"
    k8s.node.name: "localhost.localdomain"
    k8s.pod.name: "init-app-container"
    k8s.pod.uid: "c0efec20-d68c-457a-83ef-03feab6b80af"
    os.type: "linux"
    pod.creation_timestamp: "2022-08-02T13:59:34Z"
    sf_backfilledBeforeCreation: false
    sf_createdOnMs: 1659448748000L
    sf_extrapolation: "LAST_VALUE_EXTRAPOLATION"
    sf_id: FZKTQx1AgAU
    sf_isActive: true
    sf_isPreQuantized: false
    sf_key: HashSet of 10: (k8s.container.name, k8s.namespace.name, k8s.node.name, k8s.pod.name, k8s.cluster.name, sf_metric, k8s.pod.uid, host.name, os.type, container.id)
    sf_metric: "container.filesystem.capacity"
    sf_metricType: "GAUGE"
    sf_mtsCategoryType: 2L
    sf_nativeRollup: "AVERAGE_ROLLUP"
    sf_organizationID: CfhbPnRAcAM
    sf_tags: HashSet of 1: (kubernetes_service_kubernetes)
    sf_type: "MetricTimeSeries"
```